components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,169 @@
+import { Dropdown } from "components";
+import { grey800, white } from "constants/colors";
+import { addOpacityToHsl } from "helpers";
+import { usePaginationContext } from "hooks";
+import PreviousPage from "public/assets/icons/left-chevron.svg";
+import NextPage from "public/assets/icons/right-chevron.svg";
+import styled from "styled-components";
+import { DropdownItemT, PaginateForT } from "types";
+
+export interface Props {
+  paginateFor: PaginateForT;
+  numberOfEntries: number;
+}
+
+export function Pagination({ paginateFor, numberOfEntries }: Props) {
+  const { pageStates, goToPage, nextPage, previousPage, firstPage, lastPage, setResultsPerPage } =
+    usePaginationContext();
+
+  const { pageNumber, resultsPerPage } = pageStates[paginateFor];
+  const numberOfPages = Math.ceil(numberOfEntries / resultsPerPage);
+  const lastPageNumber = numberOfPages;
+  const defaultNumberOfButtons = 4;
+  const hasMorePagesThanButtons = numberOfPages >= defaultNumberOfButtons;
+  const showFirstButton = hasMorePagesThanButtons;
+  const showLastButton = hasMorePagesThanButtons;
```

Comment:
whats the meaning of `showFirstButton` `showLastButton`, i cant figure out if this logic is correct, maybe spot check this